staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -139,6 +139,33 @@ func NewAvailableConditionController(
 	return c
 }
 
+func setConditionAndUpdateStatus(client apiregistrationclient.APIServicesGetter, apiService *apiregistration.APIService, newCondition apiregistration.APIServiceCondition) error {
+	orig := apiService.DeepCopy()
+	apiregistration.SetAPIServiceCondition(apiService, newCondition)
+	if !reflect.DeepEqual(orig, apiService) {
+		_, err := client.APIServices().UpdateStatus(apiService)
+		if err != nil {
+			return err
+		}
+
+		// update metrics
+		isAvailable := apiregistration.IsAPIServiceConditionTrue(apiService, apiregistration.Available)
+		if isAvailable {
+			unavailableGauge.WithLabelValues(apiService.Name).Set(1.0)
```

Comment:
Is this backwards? Shouldn't we be incrementing the unavailableGauge when the service is not available not vice versa?

Reply:
yep. fixed.